client/admin/chat-channels.tsx

```diff
@@ -0,0 +1,242 @@
+    margin-left: 16px;
+  }
+`
+
+const StyledChannelIcon = styled(ChannelIcon)`
+  width: 56px;
+  height: auto;
+  flex-shrink: 0;
+`
+
+const ChannelInfoContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  margin: 0 8px;
+  overflow: hidden;
+`
+
+const ChannelInfoName = styled(Headline6)`
+  ${singleLine};
+`
+
+const ChannelHeadline = styled.div`
+  ${headline6};
+  margin-bottom: 8px;
+`
+
+const ChannelContainer = styled.div`
+  width: 100%;
+  overflow: hidden;
+
+  flex-grow: 1;
+  display: flex;
+`
+
+const StyledMessageList = styled(MessageList)`
+  flex-grow: 1;
+  max-width: 960px;
+  min-width: 320px;
+`
+
+export function AdminChatChannels() {
+  const dispatch = useAppDispatch()
+  const [channels, setChannels] = useState<ChannelInfo[]>()
+  const [currentPage, setCurrentPage] = useState(-1)
+  const [hasMoreChannels, setHasMoreChannels] = useState(true)
+
+  const [isLoadingMoreChannels, setIsLoadingMoreChannels] = useState(false)
+  const [searchError, setSearchError] = useState<Error>()
+  const [searchQuery, setSearchQuery] = useLocationSearchParam('q')
+  const abortControllerRef = useRef<AbortController>()
+
+  const carouselRef = useRef(null)
+  const debouncedSearchRef = useRef(
+    debounce((searchQuery: string) => {
+      // Just need to clear the search results here and let the carousel initiate the network
+      // request.
+      setSearchQuery(searchQuery)
+      setSearchError(undefined)
+      setChannels(undefined)
+      setCurrentPage(-1)
+      setHasMoreChannels(true)
+      if (carouselRef.current) {
+        // TODO(2Pac): Fix the types once the carousel is ported to TypeScript.
+        ;(carouselRef.current as any).reset()
+      }
+    }, 100),
+  )
+
+  const onSearchChange = useStableCallback((searchQuery: string) => {
+    debouncedSearchRef.current(searchQuery)
+  })
+
+  const onLoadMoreChannels = useStableCallback(() => {
+    setIsLoadingMoreChannels(true)
+    setCurrentPage(currentPage + 1)
+
+    abortControllerRef.current?.abort()
+    abortControllerRef.current = new AbortController()
+
+    fetchJson<ChannelInfo[]>(
+      apiUrl`admin/chat/?q=${searchQuery}&limit=${SEARCH_CHANNELS_LIMIT}&page=${currentPage + 1}`,
+      { signal: abortControllerRef.current.signal },
+    )
+      .then(data => {
+        setIsLoadingMoreChannels(false)
+        setChannels((channels ?? []).concat(data))
+        setHasMoreChannels(data.length >= SEARCH_CHANNELS_LIMIT)
+      })
+      .catch(err => {
+        setIsLoadingMoreChannels(false)
+        setSearchError(err)
+      })
+  })
+
+  const [selectedChannel, setSelectedChannel] = useState<ChannelInfo>()
+  const channelMessages = useAppSelector(s =>
+    selectedChannel ? s.chat.idToMessages.get(selectedChannel.id) : undefined,
+  )
+
+  const onLoadMoreMessages = useStableCallback(() => {
+    if (selectedChannel) {
+      dispatch(getMessageHistory(selectedChannel.id, CHANNEL_MESSAGES_LIMIT))
```

Comment:
I would very much prefer that you don't call things here that mess with the reducer data we use for normal chat stuff (here and the user list below). I also don't think you've actually tested any of this realistically, because the service throws errors if the user is not in the channel specified, and the whole point of this UI is that the admin doesn't need to be in the channel, so please actually test this code.